janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -222,6 +223,25 @@ public void removeConfigurationShouldRemoveGraphFromCache() throws Exception {
         }
     }
 
+    @Test
+    public void dropGraphShouldRemoveGraphFromCache() throws Exception {
```

Comment:
I just added test in it pass without changes which i expected not to happen.

Reply:
Yeah, I add a test for `drop()` because I didn't found one and wanted to check that my changes don't introduce any regressions.

I'm not sure how I can test this particular case using the `inmemory` backend, I still can reproduce the bug but there are no measurable effects. We probably could use BerkeleyDB to check this case but that involves adding a dependency to _janusgraph-test_ target.

Reply:
Valid point

Reply:
It would be okay to add BerkeleyDB to this project. I think a CQL we better so we could also test distributed database in future.

Reply:
This was the original intention before adding `CassandraContainer` to the test. Now the actual fix is checked with the [assertNull(graph_keyspace);]( https://github.com/JanusGraph/janusgraph/pull/2148/files#diff-6d6dec4c758c73971b028a06681ea37f2c7e63e828229bdd1e4d7c896e1fe409R266) in this same method.